### ISSUE TITLE
Blender hangs when start/stopping IPR

### ISSUE THREAD

**Issue by davmedei** [bsavery/PRMan-for-Blender#208]
**New problem: Blender hangs when start/stopping IPR.**

Blender hangs when stopping IPR. This is slightly different than the problem I've been experiencing before. Rendering is fine, but IPR still giving me some problems. I'm using the latest build of Blender 2.77a and Renderman 20.9. I'm using the most current devel "build" of PRMan as of May 30.

I hit start IPR, image tool opens and everything is working as intended. I can stop IPR and begin editing the scene. Image tool is still open, I click start IPR and the image tool shows no updates to the changes I've made. I click stop IPR and that is when Blender hangs. I've waited about 10 minutes with no recovery from hang.

The work around to this is that I have to close image tool before clicking Start IPR again. If I forget to do that Blender hangs and I have to force close.

I was able to catch some errors when the hang occurs:

![cmdprompt1](https://cloud.githubusercontent.com/assets/17586937/15667928/7848ac52-26b4-11e6-8623-73d025197713.PNG)

![cmdprompt2](https://cloud.githubusercontent.com/assets/17586937/15667930/7b77d79a-26b4-11e6-8c53-c8cdf51168fe.PNG)

And here are my environment vars and user prefs:

![blenderuserprefs](https://cloud.githubusercontent.com/assets/17586937/15667941/8d2b373e-26b4-11e6-8a9d-ad4ce9df71f5.PNG)

![environmentvariable](https://cloud.githubusercontent.com/assets/17586937/15667943/8f2e311c-26b4-11e6-9e80-95458fb3b5a7.PNG)

> > **Old Problem: Blender closes without warning when stopping interactive rendering**
> > 
> > Trying out Blender and Renderman together, this is the only problem I've encountered and I've tried reinstalling both Renderman and Blender.
> > 
> > I'm currently using Blender 2.76b and Renderman 20.7 Non-Commercial.
> > 
> > Here is my Renderman prefs:
> > 
> > ![capture](https://cloud.githubusercontent.com/assets/17586937/13452918/62626e6a-dff0-11e5-8ba4-8abb4a9d5d56.PNG)
> > 
> > ![captire2](https://cloud.githubusercontent.com/assets/17586937/13452924/71dac4d2-dff0-11e5-8919-916f36200a2e.PNG)
> > 
> > Blender file:
> > 
> > https://drive.google.com/open?id=0B7OPxnKNMbiTa1hoVGxhbGdpR0E
> > 
> > ____________
> > 
> > Another problem: Hitting render and then stopping a render this pops up:
> > 
> > ![screenshot 2](https://cloud.githubusercontent.com/assets/17586937/13452964/d116b514-dff0-11e5-8c5e-0329cb77d659.png)
>

**Comment by bsavery** [comment]
There sounds like there might be an issue starting "it" (the image tool)
from the Renderman studio folder.  If you start that before rendering do
you still see the problem?

Brian

On Tue, Mar 1, 2016 at 11:02 PM, davmedei <notifications@github.com> wrote:

> Problem: Blender closes without warning when stopping interactive rendering
>
> Trying out Blender and Renderman together, this is the only problem I've
> encountered and I've tried reinstalling both Renderman and Blender.
>
> I'm currently using Blender 2.76b and Renderman 20.7 Non-Commercial.
>
> Here is my Renderman prefs:
>
> [image: capture]
> <https://cloud.githubusercontent.com/assets/17586937/13452918/62626e6a-dff0-11e5-8ba4-8abb4a9d5d56.PNG>
>
> [image: captire2]
> <https://cloud.githubusercontent.com/assets/17586937/13452924/71dac4d2-dff0-11e5-8919-916f36200a2e.PNG>
>
> Blender file:
>
> https://drive.google.com/open?id=0B7OPxnKNMbiTa1hoVGxhbGdpR0E
> ------------------------------
>
> Another problem: Hitting render and then stopping a render this pops up:
>
> [image: screenshot 2]
> <https://cloud.githubusercontent.com/assets/17586937/13452964/d116b514-dff0-11e5-8c5e-0329cb77d659.png>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bsavery/PRMan-for-Blender/issues/208>.
>



-- 
brian.savery@gmail.com
508-274-8700

**Comment by davmedei** [comment]
Opened "it" before opening Blender, problem still there. I did a screen capture/recording of what happens:

https://drive.google.com/open?id=0B7OPxnKNMbiTYXA2OGNMdlZ2ZE0

**Comment by bsavery** [comment]
Hmmm ok.  You should be fine using the "Stop Interactive Rendering" Button in the blender interface though?  Rather than the one in 'it'.  I suppose should be checking if the prman process has been stopped externally in Blender.

**Comment by davmedei** [comment]
Cool, thanks, seems to work for me using just the Blender interface. 

I'm so used to the Maya setup that I figured hitting cancel render in image tool would do the same.

When I start a interactive render Blender seems to hang / stop responding for about 10 seconds and comes back. Stopping interactive render, another 10 second unresponsive Blender. Is this normal? I can instantly start and stop a "regular" render. (I have image tool always open when working)

**Comment by bsavery** [comment]
I think we could get around this by launching the initial render of PRMan in a separate thread right now it's waiting for the initial render to finish for various reasons.  Anyway I think that's a reasonable thing to look at working on!

Thanks for the interest.

**Comment by 31Feesh** [comment]
Hello, I've been experimenting with using PRMan and Blender as well, and I've been having a similar problem. I'm using RenderMan 20.8, and Blender 2.77a. Just like davmedei, I have reinstalled both Blender and RenderMan, and a fresh install of both works perfectly except for the following poblem: I've only been able to hit 'Render' once per session. 

Neither the interactive renderer nor regular renderer allow me to re-render, as Blender immediately disappears without a trace. I am able to start and stop the renderer exactly once per Blender session. 

To render again, even after NO changes in Blender (clicking nothing else besides Render after having already started or even finished rendering), I have to exit Blender and then reload the project file. Starting 'it' before starting Blender does not help (just as with davmedei), nor does starting Blender, then starting 'it'. As a matter of fact, I have not noticed 'it' having any affect at all on this particular issue.

Unfortunately, using the Blender interface does not seem to have any affect either.

I'm sure there's more information that you'd like me to give to help you diagnose the problem, but I'm not sure what that might be at the moment.

I'd appreciate any help, because if I can fix this one individual shortcoming, RenderMan will permanently be set as my default renderer!

**Comment by bsavery** [comment]
So normally what this means is the Ri context (rib gen part) isn't finishing correctly.  Is there any way you can start Blender from the command line?  Some error might get written out there that might help you diagnose? Or after the first render check the blender error console?

**Comment by davmedei** [comment]
I've updated the first post with a new problem I have with IPR.

**Comment by davmedei** [comment]
Also currently using Maya 2016 Ext 2 with RenderMan 20.9 and the same IPR lockup bug is present there too. 

The same steps done, starting IPR with IT already open causes it to lockup and become unresponsive. I've posted this to the RenderMan forums to see if anyone else has encountered this problem. 

I've already tried uninstalling and reinstalling RenderMan 20.9. So I think the problem lies within RenderMan itself rather than its plugins for Maya and Blender.

Gonna keep at it to see if I can get Maya/Blender to give me any errors that may help.
